modal/_asgi.py

```diff
@@ -15,12 +15,72 @@
 FIRST_MESSAGE_TIMEOUT_SECONDS = 5.0
 
 
-def asgi_app_wrapper(asgi_app, function_io_manager) -> Callable[..., AsyncGenerator]:
+class LifespanManager:
+    startup: asyncio.Future
+    shutdown: asyncio.Future
+    queue: asyncio.Queue
+    has_run_init: bool = False
+
+    def __init__(self, asgi_app, state):
+        self.asgi_app = asgi_app
+        self.state = state
+
+    async def ensure_init(self):
```

Comment:
Should this be an async function? It doesn't do any `await` operations.

Reply:
Added a comment now but this has to run inside the event loop which is why it's async even though it doesn't do any awaits